### PR TITLE
Build ARM/ARM64 binaries in Docker

### PR DESCRIPTION
**Pull request by webvictim**
It occurred to me that with a bit of work, we can actually cross-compile ARM and ARM64 binaries in a Docker container rather than needing to use native ARM buildboxes. This will speed up builds (Raspberry Pi builds are quite slow) and further standardise our build process. It will also allow ARM/ARM64 binaries to be built locally for testing.

I tested the binaries by building them in Docker, then copying them to an ARMv7/ARM64 machine respectively and running `teleport version`. I also checked the output of `ldd ./teleport` for versions compiled in Docker and natively - they match.

This should also open up the ability to build other ARM/ARM64-related artifacts in Docker too:
- RPM/DEB packages (#4641)
- Docker images (#3384)

This PR also:
- fixes renaming logic for OSS/Enterprise binaries in Makefiles to prevent the need for external scripts/processes to rename them
  - this is a bug that's been around for a long time, I've been meaning to fix it for a while.
- does the groundwork to add ARM64 FIPS builds (#5068)
  - changes will still need to be made to Houston to handle/serve these
- removes surplus `ADDFLAGS` set in the build.assets `Dockerfile`

Other notes:
Regarding #5482, there is currently [a bug in iovisor/gobpf](https://github.com/iovisor/gobpf/issues/272) which prevents it from being compiled on ARM, so this PR also adds conditional logic to disable BPF support in ARM binaries. We can re-enable it when the bug is fixed, although I don't imagine it'll be backported to 0.11 so we'll have to figure out what to do there. We may have to patch our own fork of the repo.

Also, I know that `.drone.yml` is a bit of a nightmare to review as there's so much boilerplate. Now that we basically have one common way to build most binaries, I'm going to start looking at ways to either offload more of the logic into the Makefile, or use a hashed/signed build script which is checked into the repo and has its checksum in the YAML file.

Depends on https://github.com/gravitational/teleport.e/pull/209

### CODE REVIEW

File: build.assets/Dockerfile-arm [hunk 1]
```diff
@@ -0,0 +1,6 @@
+ARG RUNTIME
+FROM quay.io/gravitational/teleport-buildbox:$RUNTIME
+
+RUN apt-get -y update && \
+    apt-get -y install gcc-arm-linux-gnueabihf && \
+    apt-get -y autoclean && apt-get -y clean
```
[the remark this answers is not in the document]
Comment (webvictim) [reply build.assets/Dockerfile-arm:6]: I installed both on my local Raspberry Pi without trouble, I think. I should be able to combine the Dockerfiles.

File: build.assets/Makefile [hunk 2]
```diff
@@ -9,7 +9,7 @@ HOSTNAME=buildbox
 SRCDIR=/go/src/github.com/gravitational/teleport
 DOCKERFLAGS := --rm=true -v "$$(pwd)/../":$(SRCDIR) -v /tmp:/tmp -w $(SRCDIR) -h $(HOSTNAME)
 BCCFLAGS := -v "$$(pwd)/bcc:/usr/include/bcc"
-ADDFLAGS=-ldflags -w
+ADDFLAGS ?=
```
[the remark this answers is not in the document]
Comment (webvictim) [reply build.assets/Makefile:12]: They're already [set in the regular Makefile](https://github.com/gravitational/teleport/blob/master/Makefile#L32) which means they get duplicated. If you look at the output of `make -C build.assets release` you'll see the command run is actually `GOOS=linux GOARCH=amd64 CGO_ENABLED=1 go build -tags "pam  bpf" -o build/teleport -ldflags -w -ldflags '-w -s' ./tool/teleport` which seems a bit redundant.

File: build.assets/Makefile [hunk 3]
```diff
@@ -123,6 +126,39 @@ buildbox-centos6-fips:
 		--cache-from $(BUILDBOXCENTOS6FIPS) \
 		--tag $(BUILDBOXCENTOS6FIPS) -f Dockerfile-centos6-fips .
 
+#
+# Builds a Docker buildbox for ARM builds
+#
+.PHONY:buildbox-arm
+buildbox-arm: buildbox
+	docker build \
```
[the remark this answers is not in the document]
Comment (webvictim) [reply build.assets/Makefile:134]: Doing this results in a warning from Docker: `[Warning] One or more build-args [UID GID] were not consumed`

I can add them into the Dockerfile for ARM, but given that we're using the pre-existing buildbox image which already has the UID/GID consumed it seems a little unnecessary. I can add a comment as to why we don't use it?

[the remark this answers is not in the document]
Comment (webvictim) [reply build.assets/Makefile:135]: Added a comment detailing why UID/GID aren't needed.

**Comment by webvictim** [comment]
I've made the requested changes (as well as some others) and then tested all the Drone changes with a tag build and push builds, so I'm confident this is working as intended.

**Review by awly — APPROVED** [review]

**Review by russjones — APPROVED** [review]
Bot.

**Pull request merged**